warpgate-common/src/config/mod.rs

```diff
@@ -295,26 +299,38 @@ impl WarpgateConfig {
         &self,
         for_request: Option<&poem::Request>,
     ) -> Result<Url, WarpgateError> {
-        let url = if let Some(value) = for_request.and_then(|x| x.header(http::header::HOST)) {
-            let value = value.to_string();
-            let mut url = Url::parse(&format!("https://{value}/"))?;
-            if let Some(value) = for_request.and_then(|x| x.header("x-forwarded-proto")) {
-                let _ = url.set_scheme(value);
-            }
-            url
-        } else {
-            let ext_host = self.store.external_host.as_deref();
-            let Some(ext_host) = ext_host  else {
-            return Err(WarpgateError::ExternalHostNotSet);
-          };
-            let mut url = Url::parse(&format!("https://{ext_host}/"))?;
-            let ext_port = self.store.http.listen.port();
-            if ext_port != 443 {
-                let _ = url.set_port(Some(ext_port));
-            }
-            url
+        // if trust x-forwarded, get x-forwarded-host, then try Host, then fallback on external_host
+        // if trust x-forwarded, get x-forwarded-proto, then try request scheme, then fallback https
+        // if trust x-forwarded, get x-forwarded-port, then try request port, then fallback http listen port
+        let trust_forwarded_headers = self.store.http.trust_x_forwarded_headers;
+        let url = self.store.external_host.as_ref().map(|x| Url::parse(&format!("https://{}/", x))).and_then(|x| x.ok());
+        let (scheme, host, port) = url
+            .map_or(
```

Comment:
Struggling to unterstand this line - aren't both branches the same? Did you intend `url` to be something else here?

Reply:
I realise now that it's probably slightly redundant, as the `Url::parse` will only fail if the warpgate.yaml contains errors. The `map_or` just fallsback in that case, pretty much the sasme as the url parse default values.